subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_usb.c

```diff
@@ -0,0 +1,681 @@
+		return;
+	}
+
+	ec_host_cmd_enable(ctx->c_data);
+}
+
+__maybe_unused static struct usbd_class_api ec_host_cmd_api = {
+	.request = ec_host_cmd_request,
+	.suspended = ec_host_cmd_suspended,
+	.resumed = ec_host_cmd_resumed,
+	.enable = ec_host_cmd_enable,
+	.disable = ec_host_cmd_disable,
+	.get_desc = ec_host_cmd_get_desc,
+	.init = ec_host_cmd_usbd_init,
+};
+
+static int ec_host_cmd_backend_init(const struct ec_host_cmd_backend *backend,
+				    struct ec_host_cmd_rx_ctx *rx_ctx,
+				    struct ec_host_cmd_tx_buf *tx)
+{
+	struct ec_host_cmd_usb_ctx *ctx = backend->ctx;
+	struct usbd_class_data *const c_data = ctx->c_data;
+	struct net_buf *buf;
+
+	ctx->rx_ctx = rx_ctx;
+	ctx->tx_buf = tx;
+
+	ctx->state = USB_EC_HOST_CMD_STATE_DISABLED;
+
+	k_work_init_delayable(&ctx->reset_work, ec_host_cmd_reset);
+
+	if ((ctx->rx_ctx->buf == NULL) || (ctx->tx_buf->buf == NULL)) {
+		LOG_ERR("Buffers not provided");
+		return -EINVAL;
+	}
+
+	/* EPs addresses can be changed, but it is updated in the enable procedure. */
+	buf = ec_host_cmd_buf_alloc(ec_host_cmd_get_in_bulk_ep(c_data), ctx->tx_buf->len_max,
+				    ctx->tx_buf->buf);
+	if (buf == NULL) {
+		LOG_ERR("Failed to allocate buf IN");
+		return -ENOMEM;
+	}
+	ctx->usb_tx_buf = buf;
+
+	return 0;
+}
+
+static int ec_host_cmd_backend_send(const struct ec_host_cmd_backend *backend)
+{
+	struct ec_host_cmd_usb_ctx *ctx = backend->ctx;
+	struct usbd_class_data *const c_data = ctx->c_data;
+	struct net_buf *buf = NULL;
+	int ret;
+
+	if (!atomic_test_bit(&ctx->class_state, EC_HOST_CMD_CLASS_ENABLED)) {
+		LOG_ERR("Class not enabled");
+		return -EACCES;
+	}
+
+	if (ctx->state != USB_EC_HOST_CMD_STATE_PROCESSING) {
+		LOG_ERR("Unexpected state when sending: %s", state_name[ctx->state]);
+		return -EACCES;
+	}
+
+	ctx->state = USB_EC_HOST_CMD_STATE_SENDING;
+
+	k_work_schedule(&ctx->reset_work, K_MSEC(IN_TRANSFER_TIMEOUT_MS));
+
+	net_buf_reset(ctx->usb_tx_buf);
+	ctx->usb_tx_buf->len = ctx->tx_buf->len;
+	ret = usbd_ep_enqueue(c_data, ctx->usb_tx_buf);
+	if (ret) {
+		LOG_ERR("Failed to enqueue EP IN: %d", ret);
+		k_work_reschedule(&ctx->reset_work, K_NO_WAIT);
+		return ret;
+	}
+
+	buf = ec_host_cmd_buf_alloc_int(ec_host_cmd_get_in_int_ep(c_data));
+	if (buf == NULL) {
+		LOG_ERR("Failed to allocate buf INT OUT: %d", ret);
+		k_work_reschedule(&ctx->reset_work, K_NO_WAIT);
+		return -ENOMEM;
+	}
+
+	/* Signal to host that response is ready. */
+	net_buf_add_u8(buf, USB_EC_HOST_CMD_IRQ_TYPE_RESP_READY);
+	buf->len = EP_INT_SIZE_SIZE;
+
+	ret = usbd_ep_enqueue(c_data, buf);
+	if (ret) {
+		net_buf_unref(buf);
+		LOG_ERR("Failed to enqueue EP INT OUT: %d", ret);
+		k_work_reschedule(&ctx->reset_work, K_NO_WAIT);
+		return ret;
+	}
+
+	return ret;
+}
+
+static const struct ec_host_cmd_backend_api ec_host_cmd_backend_api = {
+	.init = ec_host_cmd_backend_init,
+	.send = ec_host_cmd_backend_send,
+};
+
+BUILD_ASSERT(EP_BULK_SIZE <= 64, "Unsupported endpoint size.");
```

Comment:
I suggest to use `BUILD_ASSERT(!USBD_SUPPORTS_HIGH_SPEED, "High speed is not supported");` instead.